### PR TITLE
Add tuning doc and update tuning utility

### PR DESCRIPTION
**Pull request by KKyang**

### CODE REVIEW

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+.. _how-to-use-hipblaslt-tuning-utility:
+
+********************************
+hipBLASLt Tuning Utility
+********************************
+
+hipBLASLt has a simple utility that uses the existing kernel pools and searches for the best solution for a given problem size.
+
+The template.yaml
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
The template.yaml file
```

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+=================================
+
+The file ``template.yaml`` is under ``utilities``.
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
The ``template.yaml`` file can be found in the ``utilities`` directory.
```

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        AlgoMethod: "all"  # Fixed value
+        RotatingBuffer: 512  # It's recommended to set this value larger than the cache size of the GPU.
+    TuningParameters:
+        # SplitK list control parameter example
+        # SplitK: [0, 4, 8]  # [0] For disable
+    ProblemSizes:
+    - [128, 128, 128]  # M, N, K
+    CreateLogic: {}  # Fixed
+
+Run the tuning utility
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
Running the tuning utility
```

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+=================================
+
+The file ``find_exact.py`` is under ``utilities``.
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
The ``find_exact.py`` file is found in the ``utilities`` directory.
```

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+Steps to run the tuning:
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
Follow these steps to run the tuning:
```

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+1. Run ``./install.sh`` first. See :ref:`installation`.
+2. Make sure the ``MatchTable.yaml`` in ``build/release/library`` exists.
+3. execute ``python3 find_exact.py <your yaml file> <hipblaslt_root_folder>/build/release <output folder>``
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
1. Run ``./install.sh`` first. See :ref:`installation` for more details.
2. Ensure the ``MatchTable.yaml`` file exists in ``build/release/library``.
3. Run the command ``python3 find_exact.py <your yaml file> <hipblaslt_root_folder>/build/release <output folder>``
```

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+You should see the following message appears, in this case is NN FP32 tuning:
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
You should see the following message appear. The following example is for NN FP32 tuning:
```

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+.. code-block:: bash
+
+    Creating exact logic
+    --Reading matching table: <hipblaslt_root_folder>/build/release/library/MatchTable.yaml
+    --Reading bench files
+     --Found file <output folder>/0_Bench/result_NN_SSS_88x12x664.txt
+    Writing logic yaml files: 100%|███████| 1/1 [00:05<00:00,  5.69s/it]
+
+The final folder structure of the output folder is:
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
The final structure of the output folder is as follows:
```

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+.. image:: images/hipblaslt-tuning-folder-structure.png
+
+Folder ``0_Bench`` stores the raw benchmark results. Folder ``1_LogicYaml`` stores the output tuned Equality logic yaml.
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
The ``0_Bench`` folder stores the raw benchmark results. The ``1_LogicYaml`` folder stores the output, which is a tuned Equality logic yaml file.
```

Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]: Should Equality be capitalized here?

File: docs/sphinx/_toc.yml.in [hunk 10]
```diff
@@ -4,14 +4,17 @@ root: index
...
-    - file: installation    
+    - file: installation
   - caption: API reference
     entries:
     - file: datatypes
     - file: api-reference
     - file: ext-reference
     - file: ext-ops
     - file: logging-heuristics
+  - caption: Utilities
```
Comment (amd-jnovotny) [inline docs/sphinx/_toc.yml.in]: I recommend calling this caption How To so it aligns with the name of the folder

File: docs/how-to-use-hipblaslt-tuning-utility.rst [hunk 11]
```diff
@@ -0,0 +1,76 @@
+.. meta::
+   :description: A library that provides GEMM operations with flexible APIs and extends functionalities beyond the traditional BLAS library
+   :keywords: hipBLASLt, ROCm, library, API, tool
+
+.. _how-to-use-hipblaslt-tuning-utility:
+
+********************************
+hipBLASLt Tuning Utility
```
Comment (amd-jnovotny) [inline docs/how-to-use-hipblaslt-tuning-utility.rst]:
```suggestion
Using the hipBLASLt Tuning Utility
```

**Review by amd-jnovotny — COMMENTED** [review]
I left a few editorial suggestions. I would also suggest title the table of contents section How to and adding it to the Index page. Right now, the link is only available through the table of contents. The tile should probably be called How To.

File: docs/how-to-use-hipblaslt-tuning-utility.rst (hunk 9 above)
Comment (KKyang) [reply docs/how-to-use-hipblaslt-tuning-utility.rst]: Yes

File: docs/sphinx/_toc.yml.in (hunk 10 above)
Comment (KKyang) [reply docs/sphinx/_toc.yml.in]: Changed

**Review by aazz44ss — DISMISSED** [review]

**Comment by amd-jnovotny** [comment]
@KKyang Thanks for the changes. But I think I'd still add a link to this guide on the main index.rst page. You can create a new tile called "How to" for it.

File: docs/sphinx/_toc.yml.in [hunk 12]
```diff
@@ -4,14 +4,17 @@ root: index
...
-    - file: installation    
+    - file: installation
   - caption: API reference
     entries:
     - file: datatypes
     - file: api-reference
     - file: ext-reference
     - file: ext-ops
     - file: logging-heuristics
+  - caption: How To
```
Comment (amd-jnovotny) [inline docs/sphinx/_toc.yml.in]:
```suggestion
  - caption: How to
```

Comment (amd-jnovotny) [reply docs/sphinx/_toc.yml.in]: Sorry, that was my mistake when I made the suggestion. Should be "How to" and not "How To".

**Review by amd-jnovotny — DISMISSED** [review]
Just a suggestion and small correction (my fault on that one). Thanks!

**Review by aazz44ss — DISMISSED** [review]

**Review by vin-huang — APPROVED** [review]

**Review by ssuyuanchang — APPROVED** [review]

**Pull request merged**